### DISCUSSION TITLE
Routing pages in node.js

### DISCUSSION THREAD

**Discussion by dfigas13** [community/community#135092] [Programming Help]
### Body

Hello everyone,
My application is built with the following components:
next.js: "14.2.4" 
node: v.20.14.0
Typescript:  "^5.4.5"
My application is bilingual: en and de. The internalization was done using "next-intl".
I am currently creating the routing application for the pages. I followed every step of the official next.js documentation: https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts. I noticed that when I change the language, only part of the URL changes. For example, when I switch to English, I have http://....../en/about and when I switch to German, I have http://..../de/about.My goal is to have the URL change with the language, for example: in en: http://...../en/about and de: http://.../de/uberuns
without creating an additional folder with the German name uberuns

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by LeMill-Vycta** [comment]
Have you tried Dynamic Routing based on Language...create only one more additional folder, just one 😃, a [...slug] parent folder that nests the pages so it can change based on Language

**Comment by Chaitanya029** [comment]
Hi @dfigas13 

Thanks for sharing your question about routing and internationalization in your Next.js application.

It sounds like you're using next-intl for internalization and have set up bilingual support for English (en) and German (de). You've followed the official Next.js documentation for routing and pages, but you're experiencing an issue where only part of the URL changes when you switch languages.

Specifically, you want to achieve the following URL structure:

English: http://...../en/about
German: http://.../de/uberuns
Without creating an additional folder with the German name uberuns.

To achieve this, you can use Next.js's built-in support for internationalized routing. Here's a possible solution:

Create a next.config.js file with the following configuration:
js

Verify

Open In Editor
Edit
Copy code
module.exports = {
  //...
  i18n: {
    locales: ['en', 'de'],
    defaultLocale: 'en',
  },
};
In your pages directory, create a file called about.tsx with the following code:
tsx

Verify

Open In Editor
Edit
import { useIntl } from 'next-intl';

const AboutPage = () => {
  const intl = useIntl();

  return (
    <div>
      <h1>{intl.formatMessage({ id: 'about.title' })}</h1>
      {/* rest of the page content */}
    </div>
  );
};

export default AboutPage;
Create a intl.json file with the following content:
json

Verify

Open In Editor
Edit

{
  "en": {
    "about": {
      "title": "About"
    }
  },
  "de": {
    "about": {
      "title": "Über uns"
    }
  }
}
In your next.config.js file, add the following configuration:
js

Verify

Open In Editor
Edit
module.exports = {
  //...
  i18n: {
    locales: ['en', 'de'],
    defaultLocale: 'en',
    localeDomains: [
      {
        domain: 'example.com',
        defaultLocale: 'en',
      },
      {
        domain: 'example.com/de',
        locale: 'de',
      },
    ],
  },
};
This configuration tells Next.js to use the localeDomains feature, which allows you to map locales to specific domains or subdomains.

With this setup, when you navigate to http://example.com/en/about, Next.js will render the English version of the about page. When you navigate to http://example.com/de/uberuns, Next.js will render the German version of the about page, using the uberuns URL segment.

Note that you'll need to adjust the localeDomains configuration to match your actual domain and subdomain setup.

**Comment by UtsavSingh2902** [comment]
Express.js has an “app” object corresponding to HTTP. We define the routes by using the methods of this “app” object. This app object specifies a callback function called when a request is received. We have different methods of in-app objects for different types of requests. Node has many frameworks to help you to get your server up and running. The most popular is Express.js.
